### ISSUE TITLE
[Issue]: Latest 10.8.5 breaks "Next Up" on Main Page using [jellyfin-android]

### ISSUE THREAD

**Issue by vegedb** [jellyfin/jellyfin#8477]
### Please describe your bug

Updated to 10.8.5 and seems like it breaks "Next Up" on MAIN PAGE. 

No idea what specifically makes it break but not all shows are affected. But some do.

Only happens using [jellyfin-android](https://github.com/jellyfin/jellyfin-android) with external players such as MX Player and MPV (Using either yields the same problem).

Immediately downgrade to 10.8.4 and the problem went away. So I guess something broke on 10.8.5.

Funny thing is the following,
1. End the episode on 10.8.5
2. Find out there's no "Next Up" on main page
3. Refresh page, reboot server or Use other clients = still no "Next Up" 
4. NOW DOWNGRADE TO 10.8.4
5. The episode pops up on "Next Up" like MAGIC.

### Jellyfin Version

Other

### if other:

10.8.5

### Environment

```markdown
- OS:Debian
- Clients: Android Tablet S8 Tab
- Playback Method: MX Player and MPV
- Hardware Acceleration: YES
- Reverse Proxy: NO
```

### Jellyfin logs

```shell
10.8.5===================================================================
[2022-00-00 12:12:25.103] [WRN] [30] Jellyfin.Server.Middleware.ResponseTimeMiddleware: Slow HTTP Response from "http://10.5.6.8:8096/Shows/NextUp?Limit=24&Fields=PrimaryImageAspectRatio%2CDateCreated%2CBasicSyncInfo%2CPath%2CMediaSourceCount&UserId=cd4792df2cbd492982b3a5197b24c6b4&ImageTypeLimit=1&EnableImageTypes=Primary%2CBackdrop%2CBanner%2CThumb&EnableTotalRecordCount=false&DisableFirstEpisode=false&NextUpDateCutoff=2021-09-30T04%3A12%3A11.916Z&EnableRewatching=false" to "10.5.6.99" in 0:00:02.4381375 with Status Code 200
======================================================================




10.8.4===================================================================

[2022-00-00 11:49:35.684] [WRN] [51] Jellyfin.Server.Middleware.ResponseTimeMiddleware: Slow HTTP Response from "http://10.5.6.8:8096/Shows/NextUp?Limit=24&Fields=PrimaryImageAspectRatio%2CDateCreated%2CBasicSyncInfo%2CPath%2CMediaSourceCount&UserId=cd4792df2cbd492982b3a5197b24c6b4&ImageTypeLimit=1&EnableImageTypes=Primary%2CBackdrop%2CBanner%2CThumb&EnableTotalRecordCount=false&DisableFirstEpisode=false&NextUpDateCutoff=2021-09-30T03%3A49%3A22.603Z&EnableRewatching=false" to "10.5.6.99" in 0:00:02.3931363 with Status Code 200
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by dessalines** [comment]
Also broken for android-tv .

**Comment by SonGokuSSJ** [comment]
It is also broken for the Amazon Fire TV (1st Generation) official Jellyfin App.

**Comment by cvium** [comment]
Not a lot of information to go on here. Saying "it's broken" isn't very helpful.

Which episode should it have shown? Does the series have specials? Does the series have multiple sortings (eg. air or dvd order)? Which series is it even? Screenshots? Anything?

**Comment by dessalines** [comment]
For me it was all my currently watched TV series "Next Up" was missing.

So then on android TV, I'd go to the show, and click "Play Next", and it would say: "Could not find next episode to play". Downgrading fixed everything.

**Comment by SonGokuSSJ** [comment]
> Not a lot of information to go on here. Saying "it's broken" isn't very helpful.
Yes, sorry for not posting more information. I was still trying to see on which episode I was, because of the automatic library scan I was not able to start any episode at all. 

> Which episode should it have shown? 

For me it is for the Show "Case Closed (1996)" [https://www.themoviedb.org/tv/30983-case-closed](url). I do not remember the previous 2 occurrences, when it happened, but this time it happened on episode 158 (The Silent Ring Line). And it happens for every episode after that (not shown under "Next Episode" and when trying to play the next episode on the "Case Closed (1996)" item, it response with the error "File not found". 

> Does the series have specials? 

I do not have special episodes listed. I named them with the normal order. Episodes split into more files have been named like mentioned for movies which I found on reddit (as the -part1 -part2 or cd1 cd2 naming did not work) with just adding a (1) or (2) at the end of the corresponding part. But the mentioned episode 158 above is a single episode. 

> Does the series have multiple sortings (eg. air or dvd order)? 

Yes, on TMDB it has different sortings, but I named them with the initial sorting (1 Season with 399 episodes).

**Comment by brezzz1** [comment]
Out of curiosity, how can i go back to using 10.8.4? the repo only has the latest version and I couldn't get the manual install to work

tia

**Comment by vegedb** [comment]
> Out of curiosity, how can i go back to using 10.8.4? the repo only has the latest version and I couldn't get the manual install to work
> 
> tia

uninstall and install 10.8.4

**Comment by vegedb** [comment]
> I am having a similar issue. I had partially watched the following TV series: The Sopranos Letterkenny Stella Smiling Friends Seinfeld listed before. Recently (I suppose I updated yesterday night), only Sopranos and Seinfeld show up in the 'next up' section of TV Shows 'suggestions', and the entire Next up section doesn't appear on the home menu at all. After looking more into the issue to improve this post, I noticed that a few episodes of Letterkenny were marked unwatched (seemingly automatically), and after marking them correctly so that every episode up until the one I was on was marked watched, it also now appears in the 'next up' subsection of TV Shows 'suggestions'. I suppose it should be noted that Smiling Friends is missing S01E01 on jellyfin, and that Stella has untagged specials. Even considering these, Seinfeld is correctly tagged and marked watched.

Due to the randomness, stay on 10.8.4 for now.

**Comment by cvium** [comment]
If someone has the means/knowledge to manually build and test out #8501, feel free.

**Comment by jamesmurphy746** [comment]
Apologies if this is not the right place.. but may I ask.. is possible this bug may be the cause of a timeout error on next up searches? I'm having issues with next up in both androidtv client and via web.. clients timeout and not much in server log except for slow processing warning..

chrome debug on 'home' shows most content loaded in around 6 seconds except for next up which takes significantly longer..

...or should I start a new bug report?

thanks for your efforts btw ;)

**Comment by SenorSmartyPants** [comment]
More broken cases. I think I see the common thread with my shows that don't show up.

I have 2 shows with problems, but only 1 show per user.
User A doesn't see show X in next up. User B does see show X.
User B doesn't see show Y in next up.

For each user, they did not watch the earlier episode that JF has of that series.
User A didn't watch Show X S06E30. But watched all following episodes except for 2 most recent.
User B has watched all of Show X except 2 most recent.
Show X has no specials in JF. Using default ordering.

User B has not watched S36 episodes 13,14,15 of Show Y. There are no more episodes in JF until Season 49, User B has watched all S49 except the 3 most recent episodes. User B watched 3rd most recent but this did not get the show to display in Next up.
Show Y has no specials. Using default ordering.

I just marked S36 of Show Y as watched. Show Y now displays correctly in Next Up.

**Comment by fullmetal1** [comment]
@cvium I built the release-10.8.z branch and it seems that it's better, but still not working as I expect or as I've seen jellyfin devs describe the intended functionality. First if I leave the "maximum amount of days a show should stay in the 'Next Up' list" option at the default 365, presumably shows that I haven't touched in over a year disappear (which is the expected functionality, great!)
If I set the option to 1000 to see my other shows, I get what's depicted below.
![image](https://user-images.githubusercontent.com/1929779/197284487-abca001b-014d-4d88-b559-0f8c39413212.png)
Note that JoJo's Bizarre Adventure S03E01-02 are marked unwatched, as is Seinfeld S04E01-22. Also, another series, Legend of the Galactic Heroes: Die Neue These S04E04, despite being available, marked as unwatched, and all previous episodes in the library marked as watched, only shows up in the Next Up section when I click 'Next Up' on the home screen. It doesn't appear on the Next Up heading on the home screen itself. The episode just came out this week.

If I set the number to 0 (which I suspect is intended to mean 'never hide Next Up shows no matter the age', but this is undocumented in the options menu), S01E01 from multiple shows (possibly all of them. I didn't bother to check) I have not started watching appear on the next up heading on the home page.
